pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -538,20 +513,120 @@ func (lb *ListenerBuilder) buildSidecarOutboundListeners(node *model.Proxy,
 	// Now validate all the listeners. Collate the tcp listeners first and then the HTTP listeners
 	// TODO: This is going to be bad for caching as the order of listeners in tcpListeners or httpListeners is not
 	// guaranteed.
-	for _, l := range listenerMap {
-		if l.servicePort.Protocol.IsTCP() {
-			tcpListeners = append(tcpListeners, l.listener)
+	return finalizeOutboundListeners(lb, listenerMap)
+}
+
+func finalizeOutboundListeners(lb *ListenerBuilder, listenerMap map[string]*outboundListenerEntry) []*listener.Listener {
+	fallthroughNetworkFilters := buildOutboundCatchAllNetworkFiltersOnly(lb.push, lb.node)
+	listeners := make([]*listener.Listener, 0, len(listenerMap))
+	for _, le := range listenerMap {
+		l := buildListenerFromEntry(lb, le, fallthroughNetworkFilters)
```

Comment:
I am afraid this may change all the listeners if someone use envoyfilter to change the passthrough filter chains of a listener

Reply:
That's probably true. I will revert and I have another idea to stop making all of our code terribly inefficient for the 1% of users that use Envoyfilter. Will have fast path for non-envoyfilter users (in another PR).